agent/command/perf_send.go

```diff
@@ -53,30 +54,41 @@ func (c *perfSend) ParseParams(params map[string]interface{}) error {
 	return nil
 }
 
-func (c *perfSend) Execute(ctx context.Context,
-	comm client.Communicator, logger client.LoggerProducer, conf *internal.TaskConfig) error {
+func (c *perfSend) Execute(ctx context.Context, comm client.Communicator, logger client.LoggerProducer, conf *internal.TaskConfig) error {
 	if err := util.ExpandValues(c, conf.Expansions); err != nil {
 		return err
 	}
 
-	// Read the file and add the evergreen info.
+	// Read the file and add the Evergreen info.
 	filename := getJoinedWithWorkDir(conf, c.File)
 	report, err := poplar.LoadTests(filename)
 	if err != nil {
-		return errors.Wrapf(err, "problem reading tests from '%s'", filename)
+		return errors.Wrapf(err, "reading tests from '%s'", filename)
 	}
 	c.addEvgData(report, conf)
 
-	// Send data to Cedar.
+	// Send data to the Cedar and Data-Pipes services.
 	conn, err := comm.GetCedarGRPCConn(ctx)
 	if err != nil {
-		return errors.Wrap(err, "problem connecting to cedar")
+		return errors.Wrap(err, "connecting to Cedar")
 	}
+	dataPipes, err := comm.GetDataPipesConfig(ctx)
+	if err != nil {
+		return errors.Wrap(err, "getting the Data-Pipes config")
+	}
+	httpClient := utility.GetDefaultHTTPRetryableClient()
+	defer utility.PutHTTPClient(httpClient)
 	opts := rpc.UploadReportOptions{
-		Report:     report,
-		ClientConn: conn,
+		Report:              report,
+		ClientConn:          conn,
+		DataPipesHost:       dataPipes.Host,
+		DataPipesRegion:     dataPipes.Region,
+		AWSAccessKey:        dataPipes.AWSAccessKey,
+		AWSSecretKey:        dataPipes.AWSSecretKey,
+		AWSToken:            dataPipes.AWSToken,
+		DataPipesHTTPClient: httpClient,
 	}
-	return errors.Wrap(rpc.UploadReport(ctx, opts), "failed to upload report to cedar")
+	return errors.Wrap(rpc.UploadReport(ctx, opts), "uploading report to Cedar")
 }
 
 func (c *perfSend) addEvgData(report *poplar.Report, conf *internal.TaskConfig) {
```

Comment:
Hey just stalking the PR here, but by the end of the poplar-side PR we also added a `Requester` field to the report. If that could be added to the evg data and also if we could set the `Mainline` flag when the requester == RepoTrackerVersionRequester, that would be great!

Super cool if that should require a separate ticket though, just lmk.

Reply:
Good catch! Done.